pkg/flow/tracing/tracing.go

```diff
@@ -14,7 +14,7 @@ import (
 	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
 	"go.opentelemetry.io/otel/sdk/resource"
 	tracesdk "go.opentelemetry.io/otel/sdk/trace"
-	semconv "go.opentelemetry.io/otel/semconv/v1.17.0"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
```

Comment:
I'm a bit confused about semconv because we seem to import different versions depending on the files. Also in the changelog it goes only till 1.18 https://github.com/open-telemetry/opentelemetry-go/blob/v1.18.0/CHANGELOG.md.

Reply:
> we seem to import different versions depending on the files

I think `pkg/flow/tracing/tracing.go` is the only place where we import `go.opentelemetry.io/otel/semconv/`?

I agree that this is confusing. I did some digging just now, and this is what I think is happening:
* The [opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification/releases) repo is the one which publishes a spec, and this is what the "v1.17.0" was based on.
* Then the Otel team decided to move the semantic conventions to the [semantic-conventions](https://github.com/open-telemetry/semantic-conventions/releases) repo. v1.21.0 is the first semconv from that repo.
* The [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go/tree/main) repo is the Go API which implements the "opentelemetry-specification" and "semantic-conventions" for Go.

As per go.mod, we upgrade `opentelemetry-go` from v1.16.0 to v1.17.0. For each version of opentelemetry-go, there are implementations of the various semantic conventions.

So what I think needs to be done is to eyeball these changelogs for breaking changes:
* [opentelemetry-go](https://github.com/open-telemetry/opentelemetry-go/releases) for v1.17.0
* [semantic-conventions](https://github.com/open-telemetry/semantic-conventions/releases) for v1.21.0
* [opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification/releases) for v1.18.0, v1.19.0, v1.20.0

Reply:
Ah right, I was confused also because we import the `go.opentelemetry.io/collector/semconv/` with different versions also